client/my-sites/domains/domain-management/edit/wpcom-domain.jsx

```diff
@@ -114,8 +114,8 @@ class WpcomDomain extends React.Component {
 				}
 			>
 				{ isWpcomDomain
-					? this.props.translate( 'Change Site Address' )
-					: this.props.translate( 'Edit Site Address' ) }
+					? this.props.translate( 'Change site address' )
+					: this.props.translate( 'Edit site address' ) }
```

Comment:
:warning: This change will be queued for retranslation. We'll display the English text until then.